vspc/server.py

```diff
@@ -195,6 +198,24 @@ async def save_to_log(self, uuid, data):
         fpath = os.path.join(CONF.serial_log_dir, uuid)
         async with aiofiles.open(fpath, 'ab') as f:
             await f.write(data)
+        if CONF.log_file_size and os.path.getsize(fpath) > CONF.log_file_size:
+            truncated_data = await self.truncate_log_file(fpath, CONF.log_file_size)
+            await self.rewrite_file(fpath, truncated_data)
+
+    async def truncate_log_file(self, file_name, size):
+        async with aiofiles.open(file_name, "rb") as read_obj:
+            await read_obj.seek(0, os.SEEK_END)
+            pointer_location = await read_obj.tell()
+            start_point = pointer_location - size
+            await read_obj.seek(start_point)
+            return await read_obj.read()
+
+    async def rewrite_file(self, path, data):
+        tmp_path = "tmp_" + path
+        async with aiofiles.open(tmp_path, "wb") as f:
+            await f.write(data)
+        await aio_os.remove(path)
+        await aio_os.rename(tmp_path, path)
```

Comment:
This doesn't look threadsafe to me. Is it possible for another thread to write to the same file you're just replacing? Then we would lose data.